lighthouse-core/test/lib/asset-saver-test.js

```diff
@@ -97,4 +100,60 @@ describe('asset-saver helper', () => {
       });
     });
   });
+
+  describe('saveTrace', () => {
+    const traceFilename = 'test-trace-0.json';
+
+    afterEach(() => {
+      fs.unlinkSync(traceFilename);
+    });
+
+    it('correctly saves a trace with metadata to disk', () => {
+      return assetSaver.saveTrace(fullTraceObj, traceFilename)
+        .then(_ => {
+          const traceFileContents = fs.readFileSync(traceFilename, 'utf8');
+          assert.deepStrictEqual(JSON.parse(traceFileContents), fullTraceObj);
+        });
+    });
+
+    it('correctly saves a trace with no trace events to disk', () => {
+      const trace = {
+        traceEvents: [],
+        metadata: {
+          'clock-domain': 'MAC_MACH_ABSOLUTE_TIME',
+          'cpu-family': 6,
+          'cpu-model': 70,
+          'cpu-stepping': 1,
+          'field-trials': [],
+        }
+      };
+
+      return assetSaver.saveTrace(trace, traceFilename)
+        .then(_ => {
+          const traceFileContents = fs.readFileSync(traceFilename, 'utf8');
+          assert.deepStrictEqual(JSON.parse(traceFileContents), trace);
+        });
+    });
+
+    it('correctly saves a trace with multiple extra properties to disk', () => {
+      const trace = {
```

Comment:
I was nervous about saving a 256MB file on travis/appveyor, but I can try it out